File: q1/q1.cpp
#include <unordered_map>
#include <vector>
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        std::unordered_map <int, int> hash_table;
        std::vector<int> solution;
        for (int i = 0; i < nums.size(); i++){
            int target_remain = target - nums[i];
            if(hash_table.find(target_remain) != hash_table.end()){
                
                solution.push_back(hash_table.find(target_remain)->second);
                solution.push_back(i);
                return solution;
            }
            hash_table.insert(std::make_pair(nums[i], i));
        }
        return solution;
    }
};

File: q1-q9/q1/q1.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        solution = [];
        hashtable = {}
        for index, num in enumerate(nums):
            target_remain = target-num
            if target_remain in hashtable:
                return [hashtable[target_remain], index]
            hashtable[num] = index
        return []
